### PR TITLE
add toString() method to class generation

### PR DESCRIPTION
**Pull request by allanmckenzie**

### CODE REVIEW

File: integration-test/src/test/java/uk/gov/justice/generation/pojo/integration/test/ToStringIT.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    private final ClassInstantiator classInstantiator = new ClassInstantiator();
+    private final OutputDirectories outputDirectories = new OutputDirectories();
+
+    @Before
+    public void setup() throws Exception {
+        outputDirectories.makeDirectories("./target/test-generation/to/string");
+    }
+
+    @Test
+    public void shouldAnAddAToStringMethodFromAllFields() throws Exception {
```
Comment (mapingo) [inline integration-test/src/test/java/uk/gov/justice/generation/pojo/integration/test/ToStringIT.java]: Extra "An" in test name

File: integration-test/src/test/java/uk/gov/justice/generation/pojo/integration/test/ToStringIT.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+        final Class<?> studentClass = newClasses.get(0);
+
+        final Object student = classInstantiator.newInstance(studentClass, 23, "Fred", "Bloggs");
+
+        assertThat(student.toString(), is("StudentToString{age='23',firstName='Fred',lastName='Bloggs'}"));
+    }
+
+    @Test
+    public void shouldAnAddAToStringMethodFromAllFieldsPlusAdditionalProperties() throws Exception {
```
Comment (mapingo) [inline integration-test/src/test/java/uk/gov/justice/generation/pojo/integration/test/ToStringIT.java]: Extra "An" in test name

File: pojo-generator-core/src/main/java/uk/gov/justice/generation/SchemaPojoGenerator.java [hunk 3]
```diff
@@ -106,7 +106,7 @@ private PluginProvider createPluginProvider(final GeneratorConfig generatorConfi
 
         final ClassNameFactory classNameFactory = new ClassNameFactory(typeNameProvider, typeNamePluginProcessor);
         final JavaGeneratorFactory javaGeneratorFactory = new JavaGeneratorFactory(classNameFactory);
-        final PluginContext pluginContext = new PluginContext(javaGeneratorFactory, classNameFactory, generationContext.getSourceFilename());
+        final PluginContext pluginContext = new PluginContext(javaGeneratorFactory, classNameFactory, generationContext.getSourceFilename(), null);
```
Comment (mapingo) [inline pojo-generator-core/src/main/java/uk/gov/justice/generation/SchemaPojoGenerator.java]: The classModifyingPlugins argument is set to null, should they come from the PluginProvider?

File: pojo-generator-core/src/test/java/uk/gov/justice/generation/pojo/plugin/classmodifying/PluginContextTest.java [hunk 4]
```diff
@@ -46,8 +53,28 @@ public void shouldReturnSourceFilename() throws Exception {
...
                 UNSPECIFIED_CLASS_NAME_FACTORY,
-                sourceFilename);
+                sourceFilename,
+                emptyList());
 
         assertThat(pluginContext.getSourceFilename(), is(sourceFilename));
     }
+
+    @Test
+    public void shouldTrueIfTheSpecifiedPluginIsInUse() throws Exception {
```
Comment (mapingo) [inline pojo-generator-core/src/test/java/uk/gov/justice/generation/pojo/plugin/classmodifying/PluginContextTest.java]: Test name might need rewording

File: integration-test/src/test/java/uk/gov/justice/generation/pojo/integration/test/ToStringIT.java (hunk 2 above)
Comment (allanmckenzie) [reply integration-test/src/test/java/uk/gov/justice/generation/pojo/integration/test/ToStringIT.java]: fixed

**Pull request merged**